laiaaa0/assignment1/q2.cpp

```diff
@@ -1,126 +1,127 @@
 #include <iostream>
+#include <stdexcept>
 using namespace std;
-
+/*
+Check for loops
+Free all elements when list is destroyed
+Unique pointers
+*/
 template <class T>
 class ListNode{
   private:
-    T value;
-    ListNode * next;
+    T node_value_;
+    ListNode * node_next_;
   public:
     ListNode(){
-
-        this->next = NULL;
-
+        this->node_next_ = NULL;
```

Comment:
Forgot to mention before: nullptr is usually preferred to NULL, since it's typed. (and NULL == 0)